### PR TITLE
feat: Android Calendar (AppointmentStore)

### PR DESCRIPTION
**Pull request by pkar70**
GitHub Issue (If applicable): #3045

## PR Type

What kind of change does this PR introduce?
- Feature

## What is the current behavior?
No calendar (appointment store) implementation.
Also implemented AppointmentManager.ShowTimeFrameAsync (showing calendar for date).

## What is the new behavior?
Calendar store is implemented.
AppointmentStore and AppointmentManager - for Android.
Partial implementation: FindAppointmentsOptions, AppointmentProperties, Appointment.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [X] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information
Part of my app https://github.com/pkar70/MediumCalTile

Internal Issue (If applicable):

### CODE REVIEW

**Comment by pkar70** [comment]
Can I get any response/review of this PR?

**Comment by pkar70** [comment]
Is there any possibility to review/accept/merge this PR?

**Comment by jeromelaban** [comment]
Please add a samples app page that demonstrates the results of the invocations of the code your added.

**Comment by pkar70** [comment]
But it is only reader, not writer. So, without some other app, it cannot be tested (there would be no calendar entries in system).

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentManager.Android.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
+﻿#if __ANDROID__
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Appointments/AppointmentManager.Android.cs:1]:
```suggestion
#nullable enable

﻿#if __ANDROID__

```

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs [hunk 2]
```diff
@@ -0,0 +1,179 @@
+﻿#if __ANDROID__
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs:1]:
```suggestion
#nullable enable

﻿#if __ANDROID__

```

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+using System.Text;
+using System.Threading.Tasks;
+using Windows.Foundation;
+
+namespace Windows.ApplicationModel.Appointments
+{
+	public partial class AppointmentStore
+	{
+		private bool _StartTimeRequested = false; // if it should be included in UWP output result set
+		private bool _DurationRequested = false; // set if duration should be included in UWP output result
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs:14]:
```suggestion
		private bool _startTimeRequested = false; // if it should be included in UWP output result set
		private bool _durationRequested = false; // set if duration should be included in UWP output result

```

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+
+		private List<string> UWP2AndroColumnNames(IList<string> uwpColumns)
+		{
+			_StartTimeRequested = false;
+
+			List<string> androColumns = new List<string>();
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs:20]:
```suggestion
			List<string> androidColumns = new();

```

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+						break;
+				}
+			}
+			return androColumns;
+		}
+
+
+		public IAsyncOperation<IReadOnlyList<Appointment>> FindAppointmentsAsync(DateTimeOffset rangeStart, TimeSpan rangeLength, FindAppointmentsOptions options)
+			=> FindAppointmentsAsyncTask(rangeStart, rangeLength, options).AsAsyncOperation<IReadOnlyList<Appointment>>();
+		private async Task<IReadOnlyList<Appointment>> FindAppointmentsAsyncTask(DateTimeOffset rangeStart, TimeSpan rangeLength, FindAppointmentsOptions options)
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs:54]:
```suggestion
		private async Task<IReadOnlyList<Appointment>> FindAppointmentsAsyncTask(DateTimeOffset rangeStart, TimeSpan rangeLength, FindAppointmentsOptions options)
		

```

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+						androColumns.Add(Android.Provider.CalendarContract.IEventsColumns.Organizer);
+						break;
+					case "Appointment.Details":
+						androColumns.Add(Android.Provider.CalendarContract.IEventsColumns.Description);
+						break;
+				}
+			}
+			return androColumns;
+		}
+
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs:50]:
```suggestion
```

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+
+		public IAsyncOperation<IReadOnlyList<Appointment>> FindAppointmentsAsync(DateTimeOffset rangeStart, TimeSpan rangeLength, FindAppointmentsOptions options)
+			=> FindAppointmentsAsyncTask(rangeStart, rangeLength, options).AsAsyncOperation<IReadOnlyList<Appointment>>();
+		private async Task<IReadOnlyList<Appointment>> FindAppointmentsAsyncTask(DateTimeOffset rangeStart, TimeSpan rangeLength, FindAppointmentsOptions options)
+		{
+			Android.Database.ICursor _cursor = null;
+			Android.Content.ContentResolver _contentResolver = null;
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs:57]: Local variables don't need the `_` prefix

```suggestion
			Android.Database.ICursor? cursor = null;
			Android.Content.ContentResolver? contentResolver = null;
```

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+			List<Appointment> entriesList = new List<Appointment>();
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs:58]:
```suggestion
			List<Appointment> entriesList = new ();

```

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+
+			if (options is null)
+			{
+				throw new ArgumentNullException();
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs:62]:
```suggestion
				throw new ArgumentNullException(nameof(options));

```

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+			}
+
+			Android.Net.Uri.Builder builder = Android.Provider.CalendarContract.Instances.ContentUri.BuildUpon();
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs:65]:
```suggestion
			var builder = Android.Provider.CalendarContract.Instances.ContentUri.BuildUpon();

```

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs [hunk 11]
```diff
@@ -0,0 +1,179 @@
...
+
+			var androColumns = UWP2AndroColumnNames(options.FetchProperties);
+			// some 'system columns' columns, cannot be switched off
+			androColumns.Add(Android.Provider.CalendarContract.IEventsColumns.CalendarId);
+			androColumns.Add("_id");
+			androColumns.Add(Android.Provider.CalendarContract.Instances.Begin);    // for sort
+			androColumns.Add(Android.Provider.CalendarContract.Instances.End);    // we need this, as Android sometimes has NULL duration, and it should be reconstructed from start/end
+
+			// string sortMode = Android.Provider.CalendarContract.EventsColumns.Dtstart + " ASC";
+			string sortMode = Android.Provider.CalendarContract.Instances.Begin + " ASC";
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs:80]:
```suggestion
			var sortMode = Android.Provider.CalendarContract.Instances.Begin + " ASC";

```

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs [hunk 12]
```diff
@@ -0,0 +1,179 @@
...
+			int _colAllDay, _colLocation, _colStartTime, _colSubject, _colOrganizer, _colDetails, _colCalId, _colId, _colEndTime;
+			_colAllDay = _cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.AllDay);
+			_colLocation = _cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.EventLocation);
+			_colStartTime = _cursor.GetColumnIndex(Android.Provider.CalendarContract.Instances.Begin);
+			_colEndTime = _cursor.GetColumnIndex(Android.Provider.CalendarContract.Instances.End);
+			_colSubject = _cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.Title);
+			_colOrganizer = _cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.Organizer);
+			_colDetails = _cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.Description);
+			_colCalId = _cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.CalendarId);
+			_colId = _cursor.GetColumnIndex("_id");
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs:108]:
```suggestion
			var colAllDay = _cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.AllDay);
			var colLocation = _cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.EventLocation);
			var colStartTime = _cursor.GetColumnIndex(Android.Provider.CalendarContract.Instances.Begin);
			var colEndTime = _cursor.GetColumnIndex(Android.Provider.CalendarContract.Instances.End);
			var colSubject = _cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.Title);
			var colOrganizer = _cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.Organizer);
			var colDetails = _cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.Description);
			var colCalId = _cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.CalendarId);
			var colId = _cursor.GetColumnIndex("_id");

```

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs [hunk 13]
```diff
@@ -0,0 +1,179 @@
...
+
+
+			// odczytanie
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs:111]: Use english please.

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs [hunk 14]
```diff
@@ -0,0 +1,179 @@
...
+				{
+					entry.StartTime = DateTimeOffset.FromUnixTimeMilliseconds(_cursor.GetLong(_colStartTime));
+				}
+
+				if (_DurationRequested)
+				{
+					// calculate duration from start/end, as Android Duration field sometimes is null, and is in hard to parse RFC2445 format 
+					long startTime, endTime;
+					startTime = _cursor.GetLong(_colStartTime);
+					endTime = _cursor.GetLong(_colEndTime);
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs:145]:
```suggestion
					var startTime = _cursor.GetLong(_colStartTime);
					var endTime = _cursor.GetLong(_colEndTime);

```

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs [hunk 15]
```diff
@@ -0,0 +1,179 @@
...
+
+				entriesList.Add(entry);
+
+				if (!_cursor.MoveToNext())
+				{
+					break;
+				}
+			}
+
+			_cursor.Close();
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs:170]: It would be best to place this in a `finally`

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs (hunk 2 above)
Comment (pkar70) [reply src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs:1]: It cannot be nullable, as RequestStoreAsync returns AppointmentStore, not AppointmentStore?, and can return null.

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentManager.Android.cs (hunk 1 above)
Comment (pkar70) [reply src/Uno.UWP/ApplicationModel/Appointments/AppointmentManager.Android.cs:1]: It cannot be nullable, as RequestStoreAsync returns AppointmentStore, not AppointmentStore?, and can return null.

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs (hunk 15 above)
Comment (pkar70) [reply src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs:170]: added `using`

**Comment by pkar70** [comment]
Will build restart by itself, or should it be restarted manually?

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentManager.Android.cs (hunk 1 above)
Comment (jeromelaban) [reply src/Uno.UWP/ApplicationModel/Appointments/AppointmentManager.Android.cs]: Nullables don't change the API definition, so you should be able to make the change.

**Comment by pkar70** [comment]
Can this PR be reviewed/accepted?

**Comment by pkar70** [comment]
Can this PR be reviewed/accepted?

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs [hunk 16]
```diff
@@ -0,0 +1,240 @@
...
+		// I don't know if this is default everywhere, or only on my tablet...
+		private int DEFAULT_REMINDER_MINUTES = 15;
+
+		private async Task<IReadOnlyList<Appointment>> FindAppointmentsAsyncTask(DateTimeOffset rangeStart, TimeSpan rangeLength, FindAppointmentsOptions options)
+		{
+			List<Appointment> entriesList = new List<Appointment>();
+
+			if (options is null)
+			{
+				throw new ArgumentNullException("Windows.ApplicationModel.Appointments.AppointmentStore.FindAppointmentsAsyncTask, options cannot be null");
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs:70]: Don't use `ArgumentNullException`, use `InvalidOperationException` for all the file.

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs [hunk 17]
```diff
@@ -0,0 +1,240 @@
+﻿#nullable enable
+
+#if __ANDROID__
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs:3]: This is not needed.
```suggestion

```

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentManager.Android.cs [hunk 18]
```diff
@@ -0,0 +1,69 @@
...
+			if (options != AppointmentStoreAccessType.AllCalendarsReadOnly)
+			{
+				throw new NotImplementedException("AppointmentManager:RequestStoreAsync - only AllCalendarsReadOnly is implemented");
+			}
+
+			string requiredPermission = Android.Manifest.Permission.ReadCalendar;
+
+			if (!Windows.Extensions.PermissionsHelper.IsDeclaredInManifest(requiredPermission))
+			{
+				throw new Exception("AppointmentManager:RequestStoreAsync - no ReadCalendar permission declared in Manifest");
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Appointments/AppointmentManager.Android.cs:52]: Is this the exception type that UWP raises?

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentManager.Android.cs [hunk 19]
```diff
@@ -0,0 +1,69 @@
...
+
+		public static async Task<AppointmentStore> RequestStoreAsyncTask(AppointmentStoreAccessType options)
+		{
+			// UWP: AppCalendarsReadWrite, AppCalendarsReadOnly,AppCalendarsReadWrite (cannot be used without special provisioning by Microsoft)
+			// Android: Manifest has READ_CALENDAR and WRITE_CALENDAR, no difference between app/limited/full
+			// using only AllCalendarsReadOnly, other: throw NotImplementedException
+
+			if (options != AppointmentStoreAccessType.AllCalendarsReadOnly)
+			{
+				throw new NotImplementedException("AppointmentManager:RequestStoreAsync - only AllCalendarsReadOnly is implemented");
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Appointments/AppointmentManager.Android.cs:45]:
```suggestion
				throw new NotSupportedException("AppointmentManager:RequestStoreAsync - only AllCalendarsReadOnly is supported");

```

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentManager.Android.cs [hunk 20]
```diff
@@ -0,0 +1,69 @@
+﻿
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Appointments/AppointmentManager.Android.cs:1]:
```suggestion
#nullable enable﻿

```

File: src/Uno.UWP/Uno.csproj [hunk 21]
```diff
@@ -40,6 +40,14 @@
...
 		<UpToDateCheckInput Include="**\*.cs" Exclude="bin\**\*.cs;obj\**\*.cs;" Visible="False" />
 	</ItemGroup>
+	
+	<ItemGroup>
+	  <None Remove="ApplicationModel\Appointments\AppointmentProperties.Android.cs" />
+	</ItemGroup>
+	
+	<ItemGroup>
+	  <UpToDateCheckInput Remove="ApplicationModel\Appointments\AppointmentProperties.Android.cs" />
+	</ItemGroup>
```
Comment (jeromelaban) [inline src/Uno.UWP/Uno.csproj:50]: You can remove this.

File: src/SamplesApp/UITests.Shared/Windows_ApplicationModel/Appointments/AppointmentStore.xaml.cs [hunk 22]
```diff
@@ -0,0 +1,92 @@
...
+using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Navigation;
+using Windows.ApplicationModel.Appointments;
+
+namespace UITests.Windows_ApplicationModel.Appointments
+{
+	[SampleControlInfo("Windows_ApplicationModel.Appointments", "AppointmentStore")]
```
Comment (jeromelaban) [inline src/SamplesApp/UITests.Shared/Windows_ApplicationModel/Appointments/AppointmentStore.xaml.cs:23]:
```suggestion
	[SampleControlInfo("Windows.ApplicationModel.Appointments", "AppointmentStore")]
```

File: src/SamplesApp/UITests.Shared/Windows_ApplicationModel/Appointments/AppointmentStore.xaml.cs [hunk 23]
```diff
@@ -0,0 +1,92 @@
...
+		{
+			uiErrorMsg.Text = "";
+			uiOkMsg.Text = "";
+			uiSubject.Text = "";
+			uiStartTime.Text = "";
+
+			Windows.ApplicationModel.Appointments.AppointmentStore oCalendarStore;
+			try
+			{
+				oCalendarStore = await AppointmentManager.RequestStoreAsync(AppointmentStoreAccessType.AllCalendarsReadOnly);
```
Comment (jeromelaban) [inline src/SamplesApp/UITests.Shared/Windows_ApplicationModel/Appointments/AppointmentStore.xaml.cs:43]: The sample app does not have the read sample app permission. Could you add it so the sample app can work for this?

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentManager.Android.cs (hunk 18 above)
Comment (pkar70) [reply src/Uno.UWP/ApplicationModel/Appointments/AppointmentManager.Android.cs:52]: But this is test in Android - another structure of permissions.

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs (hunk 16 above)
Comment (pkar70) [reply src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs:70]: changing to 'return null', to be in pair with UWP behaviour (although harder to catch errors :) ).

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentManager.Android.cs (hunk 18 above)
Comment (pkar70) [reply src/Uno.UWP/ApplicationModel/Appointments/AppointmentManager.Android.cs:52]: changed to UnauthorizedAccessException (in similar situation in UWP)

File: src/SamplesApp/UITests.Shared/Windows_ApplicationModel/Appointments/AppointmentStore.xaml.cs (hunk 23 above)
Comment (pkar70) [reply src/SamplesApp/UITests.Shared/Windows_ApplicationModel/Appointments/AppointmentStore.xaml.cs:43]: Sorry, my error. Done.

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs [hunk 24]
```diff
@@ -0,0 +1,238 @@
...
+
+namespace Windows.ApplicationModel.Appointments
+{
+	public partial class AppointmentStore
+	{
+		// set to `true` if should be included in UWP output result set
+		private bool _startTimeRequested = false;
+		private bool _durationRequested = false;
+
+		private List<string> UWP2AndroColumnNames(IList<string> uwpColumns)
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs:18]:
```suggestion
		private List<string> ConvertWinRTToAndroidColumnNames(IList<string> winrtColumns)

```

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs [hunk 25]
```diff
@@ -0,0 +1,238 @@
...
+			int _colAllDay = cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.AllDay);
+			int _colLocation = cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.EventLocation);
+			int _colStartTime = cursor.GetColumnIndex(Android.Provider.CalendarContract.Instances.Begin);
+			int _colEndTime = cursor.GetColumnIndex(Android.Provider.CalendarContract.Instances.End);
+			int _colSubject = cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.Title);
+			int _colOrganizer = cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.Organizer);
+			int _colDetails = cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.Description);
+			int _colCalId = cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.CalendarId);
+			int _colHasAlarm = cursor.GetColumnIndex(Android.Provider.CalendarContract.IEventsColumns.HasAlarm);
+			int _colId = cursor.GetColumnIndex("_id");
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs:128]: Do not use leading `_` for local variables

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs [hunk 26]
```diff
@@ -0,0 +1,238 @@
...
+			var androColumns = UWP2AndroColumnNames(options.FetchProperties);
+			// some 'system columns' columns, cannot be switched off
+			androColumns.Add(Android.Provider.CalendarContract.IEventsColumns.CalendarId);
+			androColumns.Add("_id");
+			androColumns.Add(Android.Provider.CalendarContract.Instances.Begin);    // for sort
+			androColumns.Add(Android.Provider.CalendarContract.Instances.End);    // we need this, as Android sometimes has NULL duration, and it should be reconstructed from start/end
+
+			// string sortMode = Android.Provider.CalendarContract.EventsColumns.Dtstart + " ASC";
+			var sortMode = Android.Provider.CalendarContract.Instances.Begin + " ASC";
+			var _contentResolver = Android.App.Application.Context.ContentResolver;
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs:96]:
```suggestion
			var contentResolver = Android.App.Application.Context.ContentResolver;

```

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs [hunk 27]
```diff
@@ -0,0 +1,238 @@
...
+			var rangeEnd = rangeStart + rangeLength;
+			Android.Content.ContentUris.AppendId(builder, rangeEnd.ToUniversalTime().ToUnixTimeMilliseconds());
+			var oUri = builder.Build();
+			// it is simply: {content://com.android.calendar/instances/when/1588275364371/1588880164371}
+			if (oUri == null)
+			{
+				throw new NullReferenceException("Windows.ApplicationModel.Appointments.AppointmentStore.FindAppointmentsAsyncTask, oUri is null (impossible)");
+			}
+
+			var androColumns = UWP2AndroColumnNames(options.FetchProperties);
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs:87]:
```suggestion
			var androidColumns = UWP2AndroColumnNames(options.FetchProperties);

```

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs [hunk 28]
```diff
@@ -0,0 +1,238 @@
...
+
+			var builder = Android.Provider.CalendarContract.Instances.ContentUri?.BuildUpon();
+			if(builder == null)
+			{
+				throw new NullReferenceException("Windows.ApplicationModel.Appointments.AppointmentStore.FindAppointmentsAsyncTask, builder is null (impossible)");
+			}
+			Android.Content.ContentUris.AppendId(builder, rangeStart.ToUniversalTime().ToUnixTimeMilliseconds());
+			var rangeEnd = rangeStart + rangeLength;
+			Android.Content.ContentUris.AppendId(builder, rangeEnd.ToUniversalTime().ToUnixTimeMilliseconds());
+			var oUri = builder.Build();
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs:80]:
```suggestion
			var uri = builder.Build();

```

**Review by jeromelaban — COMMENTED** [review]
The sample is now working, thanks!

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs (hunk 4 above)
Comment (jeromelaban) [reply src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs]: Don't resolve the comment if you haven't done the change.

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs [hunk 29]
```diff
@@ -0,0 +1,238 @@
...
+				}
+			}
+			return androidColumns;
+		}
+
+		public IAsyncOperation<IReadOnlyList<Appointment>?> FindAppointmentsAsync(DateTimeOffset rangeStart, TimeSpan rangeLength, FindAppointmentsOptions options)
+			=> FindAppointmentsAsyncTask(rangeStart, rangeLength, options).AsAsyncOperation<IReadOnlyList<Appointment>?>();
+
+		// I don't know if this is default everywhere, or only on my tablet...
+		private int DEFAULT_REMINDER_MINUTES = 15;
```
Comment (jeromelaban) [inline src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs:61]: Move to the top of the file, and it's likely best to make it const.

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs (hunk 7 above)
Comment (jeromelaban) [reply src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs]: Don't close the comments you have not resolved.

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs (hunk 4 above)
Comment (pkar70) [reply src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs]: But this change was done!

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs (hunk 12 above)
Comment (pkar70) [reply src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs]: This change is also done (maybe not exactly, as not to `var name = ...` but to `int name=...`).

File: src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs [hunk 30]
```diff
@@ -0,0 +1,238 @@
...
+using Windows.Foundation;
+
+namespace Windows.ApplicationModel.Appointments
+{
+	public partial class AppointmentStore
+	{
+		// I don't know if this is default everywhere, or only on my tablet...
+		private const int DEFAULT_REMINDER_MINUTES = 15;
+
+		// set to `true` if should be included in UWP output result set
```
Comment (agneszitte-nventive) [inline src/Uno.UWP/ApplicationModel/Appointments/AppointmentStore.Android.cs:17]:
```suggestion
		// set to `true` it should be included in the UWP output result set

```

**Comment by agneszitte-nventive** [comment]
@pkar70 there is some conflicts to be resolved for this PR ;)

**Comment by pkar70** [comment]
Yes, I know. But as it seems that noone is interested in merging this PR into distribution, I stopped resolving conflicts - such resolving was done many times in the past, and it seems like this is resolving for resolving, not for any real progress. How many conflicts were genereted in last four months? :)
But OK, I would resolve it when I return home in next weekend or something like that.

**Comment by pkar70** [comment]
> You made changes, thank you, we'll review it again.
Yes, I made changes. On May 6th :)

**Comment by jeromelaban** [comment]
@pkar70 Thanks. We're processing changes as fast as we can, based on our client's priorities.

**Review by MartinZikmund — APPROVED** [review]

**Comment by pkar70** [comment]
Thanx for taking care of this PR.
